gdal/ogr/ogrsf_frmts/mitab/mitab_imapinfofile.cpp

```diff
@@ -632,3 +632,53 @@ int IMapInfoFile::TestUtf8Capability() const
 
     return CPLCanRecode("test", GetEncoding(), CPL_ENC_UTF8);
 }
+
+CPLString IMapInfoFile::NormalizeFieldName(const char *pszName) const
+{
+    CPLString   osName(pszName);
+    if( strlen( GetEncoding() ) > 0 )
+        osName.Recode( CPL_ENC_UTF8, GetEncoding() );
```

Comment:
this recoding might potentially fail if one provides a name that is not compatible of the dataset encoding. We should probably error out in such situation, shouldn't we ?